crates/ruff/src/checkers/ast/analyze/arg.rs

```diff
@@ -0,0 +1,27 @@
+use rustpython_parser::ast::{Arg, Ranged};
+
+use crate::checkers::ast::Checker;
+use crate::codes::Rule;
+use crate::rules::{flake8_builtins, pep8_naming, pycodestyle};
+
+/// Run lint rules over an [`Stmt`] syntax node.
+pub(crate) fn arg(arg: &Arg, checker: &mut Checker) {
+    if checker.enabled(Rule::AmbiguousVariableName) {
+        if let Some(diagnostic) = pycodestyle::rules::ambiguous_variable_name(&arg.arg, arg.range())
+        {
+            checker.diagnostics.push(diagnostic);
+        }
+    }
+    if checker.enabled(Rule::InvalidArgumentName) {
+        if let Some(diagnostic) = pep8_naming::rules::invalid_argument_name(
+            &arg.arg,
+            arg,
+            &checker.settings.pep8_naming.ignore_names,
+        ) {
+            checker.diagnostics.push(diagnostic);
+        }
+    }
+    if checker.enabled(Rule::BuiltinArgumentShadowing) {
+        flake8_builtins::rules::builtin_argument_shadowing(checker, arg);
+    }
+}
```

Comment:
These are just taking the `// Step 1: Analysis` blocks from the existing `checkers/ast/mod.rs` and moving them into a method.